Kernel/Syscalls/socket.cpp

```diff
@@ -93,15 +93,16 @@ int Process::sys$listen(int sockfd, int backlog)
     return socket.listen(backlog);
 }
 
-int Process::sys$accept(int accepting_socket_fd, sockaddr* user_address, socklen_t* user_address_size)
+int Process::sys$accept(int accepting_socket_fd, Userspace<sockaddr*> user_address, Userspace<socklen_t*> user_address_size)
 {
     REQUIRE_PROMISE(accept);
 
     socklen_t address_size = 0;
     if (user_address) {
         if (!validate_write_typed(user_address_size))
             return -EFAULT;
-        copy_from_user(&address_size, user_address_size);
+        if (!validate_read_and_copy_typed(&address_size, user_address_size))
+            return -EFAULT;
```

Comment:
Oh nice, we were only write-validating the `user_address` here. Good to get that fixed, even if harmless in practice.